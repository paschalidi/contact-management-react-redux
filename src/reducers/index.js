import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import reducerAddDeleteContact from './reducer-add-delete-contact'

const rootReducer = combineReducers({
  form: formReducer,
  reducerAddDeleteContact
});

export default rootReducer;
