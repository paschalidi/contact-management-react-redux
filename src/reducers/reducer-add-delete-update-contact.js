import {
  ADD_UPDATE_CONTACT,
  DELETE_CONTACT,
} from '../actions/types'

const INITIAL_STATE = {
  0: {
    id: null,
    firstName: null,
    lastName: null,
    email: null
  }
};

export default function (state = INITIAL_STATE, action) {
  switch (action.type) {
    case ADD_UPDATE_CONTACT:
      const { id, firstName, lastName, email } = action.payload;

      return Object.assign(
        {},
        state,
        { [id]: { id, firstName, lastName, email } }
      );
    case DELETE_CONTACT:
      const { identifier } = action.payload;
      let newState = Object.assign({}, state);

      delete newState[identifier];
      return newState;

    default:
      return state;
  }
}