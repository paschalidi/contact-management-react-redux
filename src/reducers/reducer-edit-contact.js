import {
  EDIT_CONTACT
} from '../actions/types'

const INITIAL_STATE = {
  id: null,
  firstName: null,
  lastName: null,
  email: null
};

export default function (state = INITIAL_STATE, action) {
  switch (action.type) {
    case EDIT_CONTACT:
      return {
        data: action.payload
      };
    default:
      return state;
  }
}