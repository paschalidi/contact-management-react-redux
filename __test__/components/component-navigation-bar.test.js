import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';

import reducers from '../../src/reducers';
import NavigationBar from '../../src/components/component-navigation-bar'

import renderer from 'react-test-renderer';

describe('NavigationBar', () => {
  const createStoreWithMiddleware = applyMiddleware()(createStore);
  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={createStoreWithMiddleware(reducers)}>
        <NavigationBar />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});