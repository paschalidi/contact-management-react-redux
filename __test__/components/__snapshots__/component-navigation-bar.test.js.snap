// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly 1`] = `
<div
  className="container-navigation-bar"
>
  <div
    className="component-navbar"
  >
    <div
      className="navigation-bar"
    >
      <nav
        className="navbar navbar-default"
      >
        <div
          className="container-fluid"
        >
          <div
            className="navbar-header"
          >
            <a
              className="navbar-brand"
              href="#"
            >
              Contact Managment
            </a>
          </div>
          <div
            className="collapse navbar-collapse"
          >
            <ul
              className="nav navbar-nav navbar-right"
            >
              <p
                className="navbar-text"
              >
                Contact number 
                <span
                  className="label label-primary"
                >
                  item number
                </span>
              </p>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
